src/Page/Blog/BlogPageLoader.php

```
@@ -0,0 +1,155 @@
+use Shopware\Storefront\Page\GenericPageLoaderInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class BlogPageLoader
+{
+    private SystemConfigService $systemConfigService;
+
+    private GenericPageLoaderInterface $genericLoader;
+
+    private EventDispatcherInterface $eventDispatcher;
+
+    private SalesChannelCmsPageLoaderInterface $cmsPageLoader;
+
+    private EntityRepositoryInterface $blogRepository;
+
+    public function __construct(
+        SystemConfigService                $systemConfigService,
+        GenericPageLoaderInterface         $genericLoader,
+        EventDispatcherInterface           $eventDispatcher,
+        SalesChannelCmsPageLoaderInterface $cmsPageLoader,
+        EntityRepositoryInterface          $blogRepository
+    ) {
+        $this->systemConfigService = $systemConfigService;
+        $this->genericLoader = $genericLoader;
+        $this->eventDispatcher = $eventDispatcher;
+        $this->cmsPageLoader = $cmsPageLoader;
+        $this->blogRepository = $blogRepository;
+    }
+
+    /**
+     * Loads the blog page data
+     * It gets article id from request
+     * It get Storefront Page's instance for given request
+     * It assigns metadata to page instance
+     * It dispatches an event to allow other extensions to modify the page instance
+     *
+     * @param Request $request
+     * @param SalesChannelContext $context
+     * @throws PageNotFoundException
+     * @throws InconsistentCriteriaIdsException
+     * @throws MissingRequestParameterException
+     * @return BlogPage
+     */
+    public function load(Request $request, SalesChannelContext $context): BlogPage
+    {
+        $articleId = $request->attributes->get('articleId');
+        if (!$articleId) {
+            throw new MissingRequestParameterException('articleId', '/articleId');
+        }
+
+        $blogEntry = $this->loadBlogEntry($articleId, $context);
```

throw new PageNotFoundException here instead of inside the loadBlogEntry, in case someone wants to use this method the get the blogEntry data